src/python/pants/backend/python/util_rules/faas.py

```diff
@@ -430,16 +430,29 @@ async def infer_runtime_platforms(request: RuntimePlatformsRequest) -> RuntimePl
         file_name = next(
             rt.file_name()
             for rt in request.runtime.known_runtimes
-            if version == (rt.major, rt.minor)
-            and (request.architecture is None or request.architecture.value == rt.architecture)
+            if version == (rt.major, rt.minor) and request.architecture.value == rt.architecture
         )
     except StopIteration:
-        # Not a known runtime, so fallback to just passing a platform
-        # TODO: maybe this should be an error, and we require users to specify
-        # complete_platforms themselves?
-        return RuntimePlatforms(
-            interpreter_version=version,
-            pex_platforms=PexPlatforms((_format_platform_from_major_minor(*version),)),
+        # Not a known runtime, so error out.
+        raise ValueError(
+            softwrap(
+                f"""\
+                Could not find a known runtime for the given Python version and machine architecture!
+
+                Python version: {version}
+                Machine architecture: {request.architecture.value}
+
+                Known runtimes:
+                {request.runtime.known_runtimes}
+
+                To fix, please generate a `complete_platforms` file for the given Python version and
+                machine architecture, or specify a runtime that is known to Pants.
+
+                You can follow the instructions at https://github.com/pantsbuild/pants/discussions/18756
```

Comment:
Hmmm, we should definitely have better docs for this process than a discussion that's not particularly well-formatted and is getting more and more out of date as we make improvements 🤔

Reply:
I can add a doc as part of the follow up?

Reply:
Yeah, totally! Don't feel obliged that you need to add it, I'm mostly just thinking out loud.

Sorry for ambiguity.